### ISSUE TITLE
Maybe.fromJust: Nothing

### ISSUE THREAD

**Issue by tssund93** [jonascarpay/apecs#37]
I'm having a very difficult time figuring out what's causing this exception.  As far as I can tell, it's coming from [here](https://github.com/jonascarpay/apecs/blob/89f98860e0cdf610b85b286d360220a41d97ecf9/apecs/src/Apecs/Stores.hs#L41), which I guess means the entity is getting double-deleted somehow.  I've tried everything I can to get this working but I'm not sure where I go from here.  Even if I check that the components exist for the entity that I'm deleting before I delete them it still throws the exception.

The relevant part of my code is [here](https://github.com/tssund93/haskellgame/blob/1c9de5dfe025a4c9f71147078d43ddbee5657dd5/src/Lib.hs#L186-L188), specifically the call to the `destroy` function.  If I comment that out, it works fine, and it only throws an exception when the function is called from `handleCollisions`, not `clampPlayer`.  Pardon the code, I'm new to both ECS and game programming in general.

**Comment by tssund93** [comment]
Never mind, figured it out! The function I was deleting all the components from was being called from inside a cmap lambda so it was trying to finish mapping through the deleted components.

**Comment by jonascarpay** [comment]
I was looking at your code yesterday and while I do agree your code was a bit unidiomatic, I think there should be some kind of error message. I'm reopening as a reminder to myself to include it.

**Comment by jonascarpay** [comment]
Fixed in 7689fa2f69637c8c952f1ae506bf5e0b2eaf61d1 on #34. Also 
```haskell
cmapM_ $ \(EnemyBullet _ _, ety) -> do
    entityExists <- exists ety $ Proxy @(EnemyBullet, Kinetic)
    when entityExists $ destroy ety $ Proxy @(EnemyBullet, Kinetic)
```
is functionally equivalent to
```haskell
cmap $ \(EnemyBullet _ _) -> Not @(EnemyBullet, Kinetic)
```
In general, with the proper use of `cmap`, you should almost never have to manipulate Entity values directly.
